### DISCUSSION TITLE
[firebase_messaging] API for Send to Topic

### DISCUSSION THREAD

**Discussion by awhitford** [firebase/flutterfire#5601] [Feature request]
While integrating _Firebase Messaging_, I was surprised to see that there isn't a *send* method.  I can _subscribe_ to a topic, but I can't easily send a message to that topic.  The documentation for [Sending Messages](https://pub.dev/packages/firebase_messaging#sending-messages) says to use `http.post`.

I would expect a *send* method along with related data structures for messages.

**Comment by iapicca** [comment]
Hi @awhitford 
could you please describe how would you implement the `send` method 
differently from the current http.post implementation 
and what would be the advantage of doing so
thank you

**Comment by awhitford** [comment]
What current method?  There is no _send_ method in the [API](https://pub.dev/documentation/firebase_messaging/latest/firebase_messaging/FirebaseMessaging-class.html).

**Comment by iapicca** [comment]
I've update my question

**Comment by awhitford** [comment]
I am admittedly a Noob with respect to Firebase Messaging.  However, _messaging_ implies sending and receiving messages.  The Firebase Messaging API has mechanisms to _receive_ messages; I am surprised that there does not seem to be corresponding methods to _send_ messages.

Relying on an HTTP POST to send a message seems odd.  Why isn't there a Dart method that wraps that logic?  The HTTP POST mechanism relies on managing the sensitive **Server Key** (from Firebase Console, project settings --> cloud messaging --> `Server key`); can that not be managed using the existing `GoogleService-Info.plist`?

I imagine a send method and corresponding data structures that would ensure valid communications.  Things like:
* an enum for priority (vs strings)
* validation on title, body, etc. to enforce limits
* validation on required fields

Right now, adding _Firebase Messaging_ as a dependency does not solve all of my core messaging needs.  It sounds like I need to roll my own `send` wrapper that calls http post against `https://fcm.googleapis.com/fcm/send` with json encoded body...  _Why isn't this just part of the library?_

Is this library truly a _Messaging_ library, or is it a _Notification_ library?

The lack of a send method seems to suggest that sending messages from a Flutter app is discouraged.  Is that true?  Should architectures instead send messages from server logic or Cloud Functions (where the Server Key can be secured)?
